.github/workflows/main.yml

```
@@ -122,11 +122,32 @@ jobs:
           tags: prairielearn/workspace-xtermjs:${{ env.COMMIT_SHA }}
 
       ######################################################################################################
+      # prairielearn/workspace-vscode-base
+      - name: Check whether workspaces/vscode-base was modified
+        run: tools/check_path_modified.sh workspaces/vscode-base workspaces_vscode_base_modified
+      - name: Build and push prairielearn/workspace-vscode-base
+        if: ${{ env.workspaces_vscode_base_modified }}
+        uses: docker/build-push-action@v6 # https://github.com/marketplace/actions/build-and-push-docker-images
+        with:
+          context: workspaces/vscode-base
+          platforms: linux/amd64,linux/arm64
+          push: ${{ env.CAN_PUSH_IMAGE == 'true' }}
+          no-cache: true
+          tags: prairielearn/workspace-vscode-base:${{ env.COMMIT_SHA }}
```

The output in the Actions logs gives us a hint as to why the build is still failing:

```
WARNING: No output specified with docker-container driver. Build result will only remain in the build cache. To push result image into registry use --push or to load image into docker use --load
```

I believe this will fix things:

```suggestion
        with:
          context: workspaces/vscode-base
          # We need to load this image into Docker so that subsequent builds can use it.
          load: true
          push: ${{ env.CAN_PUSH_IMAGE == 'true' }}
          no-cache: true
          tags: prairielearn/workspace-vscode-base:${{ env.COMMIT_SHA }}
```

---

While this seems to get past the `vscode-base` workspace, it fails on `vscode-cpp` or `vscode-python` because it can't find the base image. I wonder if this has something to do with trying to do both AMD and ARM. Advice is welcome!

---

> because it can't find the base image

I cannot find vscode-base in [Docker Hub](https://hub.docker.com/u/prairielearn?page=1&search=vscode) and there is a comment about secrets [not working in forks](https://github.com/PrairieLearn/PrairieLearn/blob/017e2b297d5a82baa1fbb81d18d18595feaf2c63/.github/workflows/main.yml#L31-L33) and `CAN_PUSH_IMAGE` is false in the ['successful build'](https://github.com/PrairieLearn/PrairieLearn/actions/runs/12663078076/job/35289037739?pr=11096#step:17:12) so maybe that is an issue?

Although I am also just as likely to be barking up the wrong tree as git actions are foreign to me but I am trying to learn 🤞🏻

---

Docker Hub is (or at least should be) irrelevant here. All these image builds are happening locally. They won't (nor should they) appear on Docker Hub until this lands on master.

@jgfoster can I push some commits to your branch to see if I can figure out what's going on here?

---

Actually, I see the following output in https://github.com/PrairieLearn/PrairieLearn/actions/runs/12662402765/job/35287267197:

> Error: buildx failed with: ERROR: docker exporter does not currently support exporting manifest lists

it looks like this is some insane restriction of buildx: https://github.com/docker/buildx/issues/59.

---

could we circumvent this with Matrix builds (https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/running-variations-of-jobs-in-a-workflow)?

---

How are you envisioning that'd work, split platforms into multiple jobs? `docker/build-push-action` takes care of multi-platform images automatically; now you'd have to re-assemble specific platform versions into a single manifest, and I really don't know if/how that's possible.

---

I opened https://github.com/docker/build-push-action/issues/1298 to ask about this.

---

I got builds working in my latest comments. It's not the prettiest thing (I had to enable the containerd image store and use a local registry to temporarily store the base image), but it works.

---

Very good. Thanks for your efforts. I'll clean up the sample later today (I had copied it from my environment and should have looked more carefully!).